addon-tools/helm-convert/README.md

```diff
@@ -140,6 +140,71 @@ spec:
     And another capturegroup (?<bar>.*) with no default.
 ```
 
+### lookupCR(s) substitution
+
+Since the helm-convert tool needs to make a template that's fully
+helm-compliant, we need to replace our custom `lookupCR` and `lookupCRs`
+template functions with something that is:
+
+- Functionally equivalent (returns an object or list of objects)
+- Fully defined by the user's values.yaml
+
+So this code finds all `lookupCR` or `lookupCRs` calls and expects to
+see a corresponding entry in the values.yaml for this template under the
+new `lookup_substitutions` section.
+
+For example, if you have a CR like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: downstream
+  namespace: default
+data:
+  {{- $objList := lookupCRs "v1" "ConfigMap" "default" "upstream" }}
+  {{- $value := "not found" }}
+  {{- if $objList }}
+    {{- $firstObj := index $objList 0 }}
+    {{- $value = index $firstObj "data" "fieldname" }}
+  {{- end }}
+  value: {{ $value | toYaml }}
+```
+
+With this in the values.yaml:
+
+```yaml
+cm:
+- lookup_substitutions:
+    lookupCRs_v1_ConfigMap_default_upstream:    
+      data:
+        fieldname: othervalue
+```yaml
+
+This will inject the data into the template as if there really was another
+ConfigMap named "upstream" with the canned data as follows:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: downstream
+  namespace: default
+data:
+  {{- $objList := ( "[{\"data\":{\"fieldname\":\"othervalue\"}}]" | fromJsonArray )}}
```

Comment:
would it be better to add these to the values file instead of embedding them directly?